crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -893,6 +894,17 @@ pub mod common_conditions {
         move |query: Query<(), With<T>>| !query.is_empty()
     }
 
+    /// Generates a [`Condition`](super::Condition)-satisfying closure that returns `true`
+    /// if there are any entities with the given component type removed.
+    pub fn any_component_removed<T: Component>() -> impl FnMut(RemovedComponents<T>) -> bool {
+        // `RemovedComponents` based on events and therefore events need to be consumed,
+        // so that there are no false positives on subsequent calls of the run condition.
+        // Simply checking `is_empty` would not be enough.
+        // PERF: note that `count` is efficient (not actually looping/iterating),
+        // due to Bevy having a specialized implementation for events.
+        move |mut removals: RemovedComponents<T>| !removals.iter().count() != 0
```

Comment:
```suggestion
        move |mut removals: RemovedComponents<T>| removals.iter().next().is_none()
```

This otherwise potentially does a full traversal.

Reply:
This contradicts the perf note above.

I would still prefer to remove the perf note and just use is_empty though.

Reply:
Oh I completely forgot we overrode the default implementations for those. Agreed on the use of `is_empty` instead.

Reply:
`is_empty` can't be used because it won't advance the reader and the condition will be triggered twice. We use the same strategy for events:
https://github.com/bevyengine/bevy/blob/3ead10a3e0ab1c23d8e4b6cc9fa8be1eef8c581f/crates/bevy_ecs/src/schedule/condition.rs#L856